### DISCUSSION TITLE
custom REST APIs in NocoDB [Feb 05]

### DISCUSSION THREAD

**Discussion by dstala** [nocodb/nocodb#5080] [Discord Threads]
> 🙋‍♂️ **thành**

**Comment by o1lab** [comment]
> 🙋‍♂️ **wkw**  
#### what do you mean by custom REST APIs? Can you share your use case?  
#  
> 🙋‍♂️ **thành**  
#### i just want know nocodb can customize or not  
#  
> 🙋‍♂️ **thành**  
#### do nocodb support  query caching ?  
#  
> 🙋‍♂️ **wkw**  
#### you can't  
#  
> 🙋‍♂️ **thành**  
#### ok i got it  
#  
> 🙋‍♂️ **thành**  
#### about this ?  
#  
> 🙋‍♂️ **wkw**  
#### we hv caching on metadata  
#  
> 🙋‍♂️ **thành**  
#### many thanks  
#  
> 🙋‍♂️ **thành**  
#### <@495466997304262658>  do u have any docs about nocodb plugins ?  
#  
> 🙋‍♂️ **wkw**  
#### those in app store?  
#  
> 🙋‍♂️ **thành**  
####   
![image0](https://cdn.discordapp.com/attachments/1060784470132805712/1060880881721557033/Screenshot_2023-01-06_at_18.21.45.png)  
#  
> 🙋‍♂️ **wkw**  
#### https://docs.nocodb.com/setup-and-usages/app-store  
#  

 [Autogenerated from discord](https://discord.com/channels/661905455894888490/1060784470132805712)  
- - - -  
#### Join [NocoDB's](http://www.nocodb.com/)  community  
[Discord](https://discord.gg/5RgZmkW) • [Twitter](https://twitter.com/nocodb) • [Reddit](https://www.reddit.com/r/NocoDB/) • [Docs](https://docs.nocodb.com/)

**Reply by FJEagle** [reply]
nocodb auto-generated apis are enough for most cases; but might in some cases,  we need some pretty complex SQL query, and hope could add custom rest api to execute customized SQL query. Could it be possible?
